### PR TITLE
ROX-22557: Count `expired-at` non-null changes

### PR DESCRIPTION
**Pull request by 0x656b694d**
## Description

A new Fleet-manager Prometheus metric, showing the total number of expired centrals.
The idea is to create an alert on a spike of such events.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources
- [ ] (If applicable) Changes to the dp-terraform Helm values have been reflected in the addon on integration environment

## Test manual

Initial run:

```console
$ curl -s http://localhost:8080/metrics | grep expired
# HELP acs_fleet_manager_expired_centrals_count total number of expired centrals
# TYPE acs_fleet_manager_expired_centrals_count gauge
acs_fleet_manager_expired_centrals_count 0
```

1. Created locally an eval central, managed by quota-management-list;
2. Patched quota-management-list config to allow 0 instances;
3. Restarted fleet-manager to trigger expiration worker;
4. Checked the fleet-manager metrics:

    ```console
    curl -s http://localhost:8080/metrics | grep expired
    # HELP acs_fleet_manager_expired_centrals_count total number of expired centrals
    # TYPE acs_fleet_manager_expired_centrals_count gauge
    acs_fleet_manager_expired_centrals_count 1
    ```

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -326,6 +330,20 @@ func UpdateCentralPerClusterCountMetric(clusterID string, clusterExternalID stri
 	centralPerClusterCountMetric.With(labels).Set(float64(count))
 }
 
+// create a new counterVec for when expiration timestamp is set to a central
+var centralExpirationSetCountMetric = prometheus.NewCounterVec(
```
Comment (stehessel) [inline pkg/metrics/metrics.go:334]: Why do you use a counter instead of a gauge? A counter will not decrease if instances are "un-expired" again.

**Review by stehessel — COMMENTED** [review]
Are you trying to define a metric for how many instances are expired in general? Or a metric that tracks the expired status of instances individually (presumably to detect flapping)?

**Comment by 0x656b694d** [comment]
> Are you trying to define a metric for how many instances are expired in general? Or a metric that tracks the expired status of instances individually (presumably to detect flapping)?

@stehessel, the plan is to define an alert on high rate of such expiration events, to spot potential mass extinction. The value of the metric doesn't tell much.

Implementing a metric, that counts the actual number of expired instances, would require a special process to periodically query DB specifically for that. Would you suggest that instead?

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (0x656b694d) [reply pkg/metrics/metrics.go:334]: Because this metric does not count the number of expired instances.

**Comment by 0x656b694d** [comment]
@stehessel, I changed the code to count the total number of expired centrals. This might make more sense indeed.

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (0x656b694d) [reply pkg/metrics/metrics.go]: No that's a different metric.

**Comment by stehessel** [comment]
> Implementing a metric, that counts the actual number of expired instances, would require a special process to periodically query DB specifically for that. Would you suggest that instead?

Yes from the metrics perspective this makes more sense imo. The issue with the previous approach is that the metrics will have a high cardinality. One time series per instance + probe instances will generate a lot of metric series.

**Pull request merged**